virttest/qemu_devices/qcontainer.py

```diff
@@ -3201,3 +3201,27 @@ def numa_hmat_cache_define_by_params(self, nodeid, params):
                     aobject=aobject, backend='hmat_type'))
 
         return hmat_cache_devs
+
+    def secret_object(self, secret_id, data, secret_format='raw', inline=True):
+        """
+        Generate secret object device with given data, will create a new file
+        when inline is False
+
+        :param secret_id: the id of the secret object
+        :param data: password, in raw or base64 format
+        :param secret_format: raw or base64
+        :param inline: append data=xxx if True and file=xxx if False
+        :return: the secret object device QObject
+        """
+        params = {'id': secret_id}
+        if secret_format != 'raw':
```

Comment:
How about add `format` in any case?  Does not depends on the QEMU default behavior.

Reply:
@yanan-fu Em...it also make sense, but I hope it could emulate the manual test, and maybe to discover the potential bz like the default one is not take effect.

Reply:
To discover the potential bz like the default one does not take effect, i would like to use:
`def secret_object(self, secret_id, data, secret_format, inline=True):`

Do not update format if no `secret_format` was specified in the calling. How do you think ?

Reply:
> To discover the potential bz like the default one does not take effect, i would like to use:
> `def secret_object(self, secret_id, data, secret_format, inline=True):`

Do you mean `def secret_object(self, secret_id, data, secret_format=None, inline=True):`? If so, I am okay, will update it.